.github/workflow/deploy.yml

```
@@ -0,0 +1,199 @@
+        echo "prev-sha=${PREV_SHA}" >> $GITHUB_ENV
+
+    - name: Put the site into maintenance mode
+      env:
+        APPLICATION_ENV: ${{ env.application-env }}
+        AWS_ACCOUNT_ID: ${{ secrets.OLD_AWS_ACCOUNT_ID }}
+        BRANCH: ${{ env.branch }}
+        PREV_SHA: ${{ env.prev-sha }}
+        TAG: ${{ env.tagged-image }}
+      run: |
+        source .github/bin/deploy.sh
+
+        MIGRATION_LOG=$(git log --no-merges --pretty=format:"- %s -- %an" \
+          "^${PREV_SHA}" HEAD -- data/migrations \
+          || echo "[Error retrieving log]")
+
+        # Put the site into maintenance mode if there are pending migrations.
+        # If there is any uncertainty, then fail safe by showing the
+        # maintenance page.
+        if [[ ! -z "${MIGRATION_LOG}" ]] || [ "${{ inputs.maintenance }}" = "true" ]; then
+          MAINTENANCE=1
+          register_task_definition
+          create_or_update_service
+        fi
+
+    - name: Deploy the site
+      if: ${{ ! inputs.maintenance }}
+      env:
+        APPLICATION_ENV: ${{ env.application-env }}
+        AWS_ACCOUNT_ID: ${{ secrets.OLD_AWS_ACCOUNT_ID }}
+        BRANCH: ${{ env.branch }}
+        TAG: ${{ env.tagged-image }}
+      run: |
+        source .github/bin/deploy.sh
+
+        register_task_definition
+        run_migrations
+        create_or_update_service
+
+    - name: Send Airbrake deployment notification
+      env:
+        AIRBRAKE_JS_PROJECT_ID: ${{ secrets.AIRBRAKE_JS_PROJECT_ID }}
+        AIRBRAKE_JS_PROJECT_KEY: ${{ secrets.AIRBRAKE_JS_PROJECT_KEY }}
+        APPLICATION_ENV: ${{ env.application-env }}
+      run: |
+        if [[ "${{ github.ref }}" == "refs/heads/production" ]]; then
+          AIRBRAKE_PROJECT_ID=${{ secrets.AIRBRAKE_PROJECT_ID }}
+          AIRBRAKE_PROJECT_KEY=${{ secrets.AIRBRAKE_PROJECT_KEY }}
+        else
+          AIRBRAKE_PROJECT_ID=${{ secrets.AIRBRAKE_DEV_PROJECT_ID }}
+          AIRBRAKE_PROJECT_KEY=${{ secrets.AIRBRAKE_DEV_PROJECT_KEY }}
+        fi
+
+        PAYLOAD=$(jq -n \
+          --arg environment "${APPLICATION_ENV}" \
+          --arg repository "https://github.com/${{ github.repository }}" \
+          --arg revision "${{ github.sha }}" \
+          '{"environment": $environment, "repository": $repository, "revision": $revision}')
+
+        curl -X POST \
+          -H "Content-Type: application/json" \
+          --data "${PAYLOAD}" \
+          "https://airbrake.io/api/v4/projects/${AIRBRAKE_PROJECT_ID}/deploys?key=${AIRBRAKE_PROJECT_KEY}"
+
+        curl -X POST \
+          -H "Content-Type: application/json" \
+          --data "${PAYLOAD}" \
+          "https://airbrake.io/api/v4/projects/${AIRBRAKE_JS_PROJECT_ID}/deploys?key=${AIRBRAKE_JS_PROJECT_KEY}"
+
+    - name: Generate deployment notification message
+      id: generate-deployment-notification-message
+      if: ${{ github.ref == 'refs/heads/production' || github.ref == 'refs/heads/release' || github.ref == 'refs/heads/test' }}
```

```suggestion
      if: ${{ github.ref == 'refs/heads/production' || github.ref == 'refs/heads/release' || github.ref == 'refs/heads/dev' }}
```

we don't have test in journal-guide